resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/repository/AbstractReactiveRepositoryRedisAdapter.java

```diff
@@ -222,44 +241,49 @@ protected AbstractReactiveRepositoryRedisAdapter(final ReactiveRedisTemplate<Str
         /**
          * {@link CompletenessRepositoryAdapter} constructor that takes a redis template
          *
-         * @param redisTemplate a {@link ReactiveRedisTemplate} with key of type {@link String} and value of type {@link CompletenessEntity}
-         * @param keyPrefix     a prefix to the table 'namespace'
+         * @param redisTemplate   a {@link ReactiveRedisTemplate} with key of type {@link String} and value of type {@link CompletenessEntity}
+         * @param redisProperties containing properties used to configure redis
          */
-        public CompletenessRepositoryAdapter(final ReactiveRedisTemplate<String, CompletenessEntity> redisTemplate, final String keyPrefix) {
-            super(redisTemplate, keyPrefix + reflectTableAnnotation(CompletenessEntity.class).value());
+        public CompletenessRepositoryAdapter(final ReactiveRedisTemplate<String, CompletenessEntity> redisTemplate, final RedisProperties redisProperties) {
+            super(redisTemplate, redisProperties.getKeyPrefix() + reflectTableAnnotation(CompletenessEntity.class).value(), redisProperties.timeToLiveFor(reflectTableAnnotation(CompletenessEntity.class).value()));
         }
 
         @Override
         public Mono<CompletenessEntity> findOneByEntityTypeAndEntityId(final EntityType entityType, final String entityId) {
-            return this.findById(CompletenessEntity.idFor(entityType, entityId));
+            return this.valueOps.get(this.table + KEY_SEP + CompletenessEntity.idFor(entityType, entityId));
         }
 
         @Override
         @NonNull
         public <S extends CompletenessEntity> Mono<S> save(final @NonNull S entity) {
-            return this.saveDefault(entity);
+            final String id = this.table + KEY_SEP + entity.getId();
+            return this.valueOps.set(id, entity)
+                    .then(this.redisTemplate.expire(id, ttl))
+                    .thenReturn(entity);
         }
 
         @Override
         @NonNull
         public Mono<Void> deleteById(final @NonNull Integer key) {
-            return this.deleteByIdDefault(key);
+            return this.valueOps.delete(this.table + KEY_SEP + key)
+                    .then();
         }
     }
 
     /**
      * A class to allow {@link ResourceEntity}s to be stored in a reactive repository
      */
     public static class ResourceRepositoryAdapter extends AbstractReactiveRepositoryRedisAdapter<ResourceEntity, String> implements ResourceRepository {
+        private static final String SEPARATOR = KEY_SEP + PARENT_KEYSPACE + KEY_SEP;
```

Comment:
I'd rename this to maybe `PARENT_SEP` or similar

Reply:
Renamed